mkosi

```diff
@@ -2736,17 +2780,30 @@ def install_grub(args: CommandLineArguments, root: str, loopdev: str, grub: str)
     kernel_cmd_line = ' '.join(args.kernel_command_line)
     grub_cmdline = f'GRUB_CMDLINE_LINUX="{kernel_cmd_line}"\n'
     os.makedirs(os.path.join(root, "etc/default"), exist_ok=True, mode=0o755)
-    if not os.path.exists(os.path.join(root, "etc/default/grub")):
-        with open(os.path.join(root, "etc/default/grub"), "w+") as f:
+    grub_config = os.path.join(root, "etc/default/grub")
+    if not os.path.exists(grub_config):
+        with open(grub_config, "w+") as f:
             f.write(grub_cmdline)
     else:
         def jj(line: str) -> str:
             if line.startswith("GRUB_CMDLINE_LINUX="):
                 return grub_cmdline
+            elif "GRUB_TERMINAL_INPUT" in line and args.qemu_headless:
+                return "GRUB_TERMINAL_INPUT=\"console serial\""
+            elif "GRUB_TERMINAL_OUTPUT" in line and args.qemu_headless:
```

Comment:
I'd switch the two `and` clauses, because if the first is false the other will not be evaluated anymore. Not that it makes much of a difference, but it should be quicker to check the truthiness if a bool that to check whether a string is a substring of another.